src/ActionUnit.cpp

```diff
@@ -502,11 +531,30 @@ TAction* ActionUnit::getHeadAction(TEasyButtonBar* pT)
 
 void ActionUnit::constructToolbar(TAction* pA, TEasyButtonBar* pTB)
 {
+    if (!pA->isDataChanged()) {
```

Comment:
This should speed things up like the corresponding bit in the other method with the same name but different signature.